src/agent/rustjail/src/container.rs

```diff
@@ -1665,21 +1661,24 @@ mod tests {
                 annotations: Default::default(),
             },
         )
-        .await
-        .unwrap();
+        .await;
 
-        assert_eq!(Path::new(&temp_file).exists(), true);
+        assert!(res.is_ok());
     }
 
     #[tokio::test]
     async fn test_execute_hook_with_error() {
-        let ls = which("ls").await;
+        let sh = which("sh").await;
 
         let res = execute_hook(
             &slog_scope::logger(),
             &Hook {
-                path: ls,
-                args: vec!["ls".to_string(), "/tmp/not-exist".to_string()],
+                path: sh,
+                args: vec![
+                    "sh".to_string(),
+                    "-c".to_string(),
+                    "sleep 1 && exit 1".to_string(),
```

Comment:
Is there any other way to deal with this? Sleeps are fragile and this is bound to fail on some (slow) h/w one day.

Reply:
@jodh-intel  I'm going to close this one and fix this in https://github.com/kata-containers/kata-containers/pull/3668